### DISCUSSION TITLE
vim can not find libcrypt.so.1

### DISCUSSION THREAD

**Discussion by cang-mang** [Homebrew/discussions#3570] [Linux]
I run `brew update` and `brew upgrade`, then the `glibc` upgrades from 2.23 to 2.35, and the `vim` upgrades from 9.0.0200 to 9.0.0250. But when I run `vim`, it says `vim: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory`. And I can not find `libcrypt.so.1` in `/home/linuxbrew/.linuxbrew/lib` and `/home/linuxbrew/.linuxbrew/Cellar/glibc/2.35/lib`.

**Comment by konnano** [comment]
It's a homebrew bug
vim doesn't work, Reproduced on my CentOS Linux8
Dependencies messed up
I did a homebrew reinstall

**Comment by danielnachun** [comment]
It will be fixed by https://github.com/Homebrew/homebrew-core/pull/108663.  In the interim, you can fix this your self by doing the following:
1) `brew install libnsl` (needed to rebuild `perl`)
2) Edit the `perl` formula to depend on `libxcrypt` with the line `uses_from_macos "libxcrypt"`
3) Reinstall `perl` with `brew reinstall -v -s perl`

At that point it should be fixed.

**Comment by danielnachun** [comment]
In case anyone finds this, it is now fixed because https://github.com/Homebrew/homebrew-core/pull/108663 is merged.
